Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-NewServiceFabricApplication.ps1

```diff
@@ -206,7 +206,8 @@
                     # It will unregister the existing application's type and version even if its different from the application being created,
                     if ((Get-ServiceFabricApplicationAction | Where-Object {$_.ApplicationTypeVersion -eq $($app.ApplicationTypeVersion) -and $_.ApplicationTypeName -eq $($app.ApplicationTypeName)}).Count -eq 0)
                     {
-                        Unregister-ServiceFabricApplicationTypeAction -ApplicationTypeName $($app.ApplicationTypeName) -ApplicationTypeVersion $($app.ApplicationTypeVersion) -TimeoutSec $UnregisterPackageTimeoutSec
+                        Unregister-ServiceFabricApplicationTypeAction -ApplicationTypeName $($app.ApplicationTypeName) -ApplicationTypeVersion $($app.ApplicationTypeVersion)
+                        Wait-ServiceFabricApplicationTypeRegistrationStatus -ApplicationTypeName $($app.ApplicationTypeName) -ApplicationTypeVersion $($app.ApplicationTypeVersion) -TimeoutSec $UnregisterPackageTimeoutSec
```

Comment:
I don't wanted that just because both the utility functions has their own retry mechanisms, hence if we move Wait-ServiceFabricApplicationTypeRegistrationStatus inside Unregister-ServiceFabricApplicationTypeAction then for the cases where where Wait-ServiceFabricApplicationTypeRegistrationStatus throw exception then it will be evaluated for Unregister-ServiceFabricApplicationTypeAction as well which can cause issues.

I'm not able to come up with any case where your suggested approach will break so I'm doing changes as suggested by you.